src/main/java/net/openhft/chronicle/network/cluster/handlers/HeartbeatHandler.java

```diff
@@ -224,24 +225,16 @@ public String toString() {
     class HeartbeatCheckHandler implements VanillaEventHandler {
         @Override
         public boolean action() throws InvalidEventHandlerException {
-
-            if (HeartbeatHandler.this.closable().isClosed())
-                throw newClosedInvalidEventHandlerException();
+            if (HeartbeatHandler.this.isClosed())
+                throw InvalidEventHandlerException.reusable();
 
             boolean hasHeartbeats = HeartbeatHandler.this.hasReceivedHeartbeat();
             boolean prev = HeartbeatHandler.this.hasHeartbeats.getAndSet(hasHeartbeats);
 
             if (hasHeartbeats != prev) {
                 if (!hasHeartbeats) {
-                    final Runnable socketReconnector = HeartbeatHandler.this.nc().socketReconnector();
-                    if (socketReconnector == null)
-                        Jvm.warn().on(getClass(), "socketReconnector == null");
-                    else
-                        socketReconnector.run();
-
-                    HeartbeatHandler.this.close();
-
-                    throw newClosedInvalidEventHandlerException();
+                    HeartbeatHandler.this.hasTimedOut = true;
+                    throw InvalidEventHandlerException.reusable();
```

Comment:
Just set `hasTimedOut` to true, and stop checking for timeouts. The rest of the flow will be taken care of by the TcpEventHandler